dora/src/stdlib.rs

```diff
@@ -249,18 +249,28 @@ pub extern "C" fn argv(ind: i32) -> Ref<Str> {
     panic!("argument does not exist");
 }
 
-pub extern "C" fn str_parse_int(val: Ref<Str>) -> i32 {
+pub extern "C" fn str_to_int_or_throw(val: Ref<Str>) -> i32 {
     let slice = val.content();
     let val = str::from_utf8(slice).unwrap();
 
-    val.parse::<i32>().unwrap_or(0)
+    let result = val.parse::<i32>();
+    if result.is_ok() {
+        return result.unwrap()
+    }
+    throw_native(true);
+    0
 }
 
-pub extern "C" fn str_parse_long(val: Ref<Str>) -> i64 {
+pub extern "C" fn str_to_long_or_throw(val: Ref<Str>) -> i64 {
     let slice = val.content();
     let val = str::from_utf8(slice).unwrap();
 
-    val.parse::<i64>().unwrap_or(0)
+    let result = val.parse::<i64>();
+    if result.is_ok() {
+        return result.unwrap()
+    }
+    throw_native(true);
```

Comment:
`throw_native` is just a test function and isn't intended to be used apart from that. The boolean argument is just there to simulate a function that sometimes throws but also succeeds. Could you please create another function that just creates the exception object and sets the flag. throw_native could also use this function when the argument is true.

Reply:
Done!